python/app/dialog.py

```diff
@@ -153,6 +162,20 @@ def __init__(self, parent=None):
         self._menu = shotgun_menus.ShotgunMenu(self.ui.action_button)
         self.ui.action_button.setMenu(self._menu)
 
+        # Last sort menu item selected, by default is sorted 'due_date'
+        self._current_menu_sort_item = "due_date"
+        self._current_menu_sort_order = "desc"
+
+        # Create a Horizontal widget layout
+        # to place the Filter and Sort menus
+        self._sort_filter_layout = QtGui.QHBoxLayout()
+        # The layout to nest the label and sort_filter layout
+        self._label_menus_layout = QtGui.QHBoxLayout()
+        # Create the filter menu
+        self._sg_filter_menu = None
+        # create the sort menu
+        self._entity_field_menu = None
```

Comment:
Currently there is only one filter and sort menu set up for Tasks, but if you wanted to add the filter/sort menus to another tab, say Versions, then we would only have one filter/sort menu for both Tasks and Versions.

In order to support multiple tabs with the filter/sort menus, these variables should be stored with the specific entity tab data.

The UI elements should just be created in the `build_entity_tabs` function (_sort_filter_layout, _label_menus_layout).